### DISCUSSION TITLE
Creating a prisma schema using a json format which contains  bulk of objects

### DISCUSSION THREAD

**Discussion by jyoti-ranjan-4** [prisma/prisma#21730] [Q&A] [answered]
### Question

Hey team,
I'm trying to collect all my data from mongodb. The data in mongodb is in below format:
```
{
  "cloud": "AZ",
  "time_stamp": "2023-10-17T07:16:31Z",
  "ABC-AZ-1": {
    "instance_name": "ABC-AZ-1",
    "instance_id": "abc/4332",
    "os": "Windows",
    "version": "18.04.202306070"
  },
  "ABC-AX-223": {
    "instance_name": "ABC-AX-12",
    "instance_id": "axc/4fdsd32",
    "os": "Linux",
    "version": "18.04.202306070"
  }
}
```

If I run `npx prisma db pull` then it gives me the following schema in `schema.prisma`:

```prisma
model Instances {
  id                         String                   @id @default(auto()) @map("_id") @db.ObjectId
  ABC-AX-223       InstanceABCAX223@map("ABC-AX-223")
  ABC-AZ-1            InstanceABCAZ1 @map("ABC-AZ-1")
  cloud                   String
  time_stamp         String
}

type InstanceABCAX223{
 instance_id                String
  instance_name              String
 os                         String
version                    String
}

type InstanceABCAZ1{
 instance_id                String
  instance_name              String
 os                         String
version                    String
}
and so on...
```

But this schema is not an optimized and consistent. In future we may add more items like `ABC-AX-223` and `ABC-AZ-1`. So it may cause problems. Can anyone have any suggestions for the above json format to make it an optimized schema?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @jyoti-ranjan-4 👋 

Thank you for raising this question.

One approach to handle the dynamic nature of your data is to use Prisma's [JSON](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#json)  data type field inside the schema and store any arbitrary data deeply nested inside the said field. This approach allows you to store unstructured data in MongoDB with arbitrary top-level fields not defined in the Prisma Schema.
Here's an example of how you could define your schema:
```prisma
model Instances {
  id          String   @id @default(auto()) @map("_id") @db.ObjectId
  cloud       String
  time_stamp  String
  data        Json
}
```

In this schema, the data field can contain any JSON data, including the instance data in your original format.
Please note that this approach has limitations. Prisma Client will not provide type safety for the data inside the Json field, and you will need to handle the data manually in your application code.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by jyoti-ranjan-4** [reply]
Thanks @ludralph for your kind help.

**Reply by ludralph** [reply]
You're welcome 🙌
